### PR TITLE
WIP configeditor: make config editor a managed component

### PR DESCRIPTION
**Pull request by nichitagutu**
This PR makes config editor a manageable component, so it can be unmanaged and deleted
([PROJQUAY-2763](https://issues.redhat.com/browse/PROJQUAY-2763))

### CODE REVIEW

**Comment by ibazulic** [comment]
What would making the config editor a managed component and subsequent removing accomplish?

**Comment by nichitagutu** [comment]
> What would making the config editor a managed component and subsequent removing accomplish?

> While password-protected, the config editor presents a security risk being exposed on the network. It's not needed for running Quay after the initial deployment nor by the operator, so it should be possible to disable the config editor or at least remove the public ingress point.
 [projquay-2763](https://issues.redhat.com/browse/PROJQUAY-2763)

My understanding is that by making config editor manageable, user will have ability to unmanage this component and delete it, so it would not pose a security risk. Correct me if I'm wrong.

File: kustomize/components/configeditor/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Clair component adds Clair v4 security scanner and its database.
```
Comment (flavianmissi) [inline kustomize/components/configeditor/kustomization.yaml]: left from copy/paste?

**Review by flavianmissi — COMMENTED** [review]
left one comment - lgtm otherwise

File: kustomize/components/configeditor/kustomization.yaml (hunk 1 above)
Comment (nichitagutu) [reply kustomize/components/configeditor/kustomization.yaml]: Indeed :/ 
Fixed

**Comment by flavianmissi** [comment]
Looks like the tests need some love as well @Nikita757

File: pkg/kustomize/kustomize_test.go [hunk 2]
```diff
@@ -241,9 +241,7 @@ func TestFlattenSecret(t *testing.T) {
 var quayComponents = map[string][]client.Object{
 	"quay": {
 		&appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "quay-app"}},
-		&appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "quay-config-editor"}},
 		&corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "quay-app"}},
-		&corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "quay-config-editor"}},
 		&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "quay-config-secret"}},
```
Comment (flavianmissi) [inline pkg/kustomize/kustomize_test.go:245]: Should this secret also be moved to the `configeditor` component?

File: apis/quay/v1/quayregistry_types.go [hunk 3]
```diff
@@ -150,6 +152,7 @@ const (
 	ComponentMirrorReady        ConditionType = "ComponentMirrorReady"
 	ComponentMonitoringReady    ConditionType = "ComponentMonitoringReady"
 	ComponentTLSReady           ConditionType = "ComponentTLSReady"
+	ComponenntConfigEditorReady ConditionType = "ComponentConfigEditorReady"
```
Comment (flavianmissi) [inline apis/quay/v1/quayregistry_types.go:155]: you have a typo in the const name, replace `ComponenntConfigEditorReady` with `ComponentConfigEditorReady`.
furthermore, where is this condition being used? maybe we need something in `pkg/cmpstatus` to control it?

**Pull request closed without merging**